lib/routers/rops.js

```diff
@@ -0,0 +1,53 @@
+const { Router } = require('express');
+const { BadRequestError } = require('@asl/service/errors');
+
+module.exports = settings => {
+  const router = Router({ mergeParams: true });
+
+  router.get('/', (req, res, next) => {
+    if (!req.query.year) {
+      throw new BadRequestError('year must be provided');
+    }
+
+    const year = req.query.year;
+    let due = 0;
+    let submitted = 0;
+    let outstanding = 0;
+
+    Promise.resolve()
+      .then(() => {
+        const q = req.db.asl('projects')
+          .select('id')
+          .where('issue_date', '<=', `${year}-12-31`)
+          .whereNull('deleted')
+          .andWhere(builder => {
+            builder
+              .where({ status: 'active' })
+              .orWhere(qb => {
+                qb
+                  .where({ status: 'expired' })
+                  .where('expiry_date', '>=', `${year}-01-01`);
+              })
+              .orWhere(qb => {
+                qb
+                  .where({ status: 'revoked' })
+                  .where('revocation_date', '>=', `${year}-01-01`);
+              });
+          });
+        return q;
+      })
+      .then(projects => {
+        due = projects.length;
+      })
+      .then(() => req.db.asl('rops').select('id', 'status').where({ year }))
+      .then(rops => {
+        submitted = rops.filter(rop => rop.status === 'submitted').length;
```

Comment:
It's probably safer to define this query as the above project query with an additional `whereExists` clause. That avoids any possible issues with ROPs ending up not perfectly one-to-one.

Reply:
Any particular reason to fetch the data then run `filter` rather than doing the filter in the database query?

Reply:
Just a remnant as I was initially counting drafts as well, I've moved it to the query but haven't pushed it yet.